### PR TITLE
updated the docs for creating a new component

### PR DESCRIPTION
**Pull request by harshad16**
updated the docs for creating a new component
Signed-off-by: Harshad Reddy Nalla <hnalla@redhat.com>

More docs updates are along the way. :train:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,32 +15,58 @@ Thoth is deployed to the following [environments](docs/environments.md).
...
 ### Developing a new component
 
+Developing a new component under thoth-station:
+
+- Please make use of the [template-project](https://github.com/thoth-station/template-project) it has basic requirements. Github Allow users to create new repository with template projects.<br>
+  If your are a new member of Team Thoth and need a repository on Thoth-station. Please create an issue in Thoth-application.
+
+- Once the Developing work is done on the repository, please make use of the Thoth-bots for smooth release. More information on updating the configuration of the Thoth-bots [click here](https://github.com/AICoE/aicoe-ci#setting-aicoe-ci-on-github-organizationrepository).
+
+- Bots also allow user to configuration them to auto update the deployment of the component in thoth-application. For this firstly, we have to setup an component for the application.<br>
```
Comment (pacospace) [inline README.md:40]: `to configure`

`a component`

File: README.md [hunk 2]
```diff
@@ -12,32 +15,58 @@ Thoth is deployed to the following [environments](docs/environments.md).
...
+  The component must be created based on the description of the project. The component has two sub-directory **base** and **overlays**, this kustomize methods which is followed by argocd applications.<br>
```
Comment (pacospace) [inline README.md:41]: `these`?

**Review by pacospace — COMMENTED** [review]
minor comments, great docs :)

File: README.md [hunk 3]
```diff
@@ -12,32 +15,58 @@ Thoth is deployed to the following [environments](docs/environments.md).
 
 ## Separation of Application Components and Application Configuration
 
-Each component of the Thoth Application is separated into an ArgoCD Application, there is a core Application
-that contains global configs and secrets.
+Each component of the Thoth Application is separated into an ArgoCD Application.<br>
```
Comment (fridex) [inline README.md:18]:
```suggestion
Each component of the Thoth Application is separated into an ArgoCD Application.

```

File: README.md [hunk 4]
```diff
@@ -12,32 +15,58 @@ Thoth is deployed to the following [environments](docs/environments.md).
...
+### Test a pull request
 
-### pushing a release to production
+### Push Release to Production
 
 ### Developing a new component
 
+Developing a new component under thoth-station:
+
+- Please make use of the [template-project](https://github.com/thoth-station/template-project) it has basic requirements. Github Allow users to create new repository with template projects.<br>
```
Comment (fridex) [inline README.md:35]:
```suggestion
- Please make use of the [template-project](https://github.com/thoth-station/template-project) it has basic requirements. Github Allow users to create new repository with template projects.

```

File: README.md (hunk 1 above)
Comment (fridex) [inline README.md:40]:
```suggestion
- Bots also allow user to configuration them to auto update the deployment of the component in thoth-application. For this firstly, we have to setup an component for the application.

```

File: README.md [hunk 5]
```diff
@@ -12,32 +15,58 @@ Thoth is deployed to the following [environments](docs/environments.md).
...
+- First, create a `secrets.yaml` at the root level. (git would ignore the file)
+- Then run sops command to encrypt them to the desired application subdirectory. Example: `sops -e secrets.yaml > {component}/overlays/test/secrets.enc.yaml`
+- Make sure to contribute the encrypted **secrets.enc.yaml** .
 
 ## Open Questions
 
-### how do we tag release
+- What should i do if i don't know how to deploy my application?
+
+  we all are happy to help, open a issue with details of the application.<br>
```
Comment (fridex) [inline README.md:63]:
```suggestion
  we all are happy to help, open an issue with details of the application.<br>
```

**Review by fridex — COMMENTED** [review]
Seen HTML br tag in the file (not all marked). Is it intentional?

Otherwise very nice! :+1:

File: README.md (hunk 1 above)
Comment (harshad16) [reply README.md]: Thanks

**Comment by harshad16** [comment]
> Seen HTML br tag in the file (not all marked). Is it intentional?
Need to change my text-editor plugin, it auto adds them for formatting :sweat:

File: README.md [hunk 6]
```diff
@@ -12,32 +15,59 @@ Thoth is deployed to the following [environments](docs/environments.md).
 
 ## Separation of Application Components and Application Configuration
 
-Each component of the Thoth Application is separated into an ArgoCD Application, there is a core Application
-that contains global configs and secrets.
+Each component of the Thoth Application is separated into an ArgoCD Application.
+The core Application that contains global configs and secrets along with the setup of the namespace based requirements.
```
Comment (pacospace) [inline README.md:19]: The core Application contains global configs and secrets along with the setup of the namespace based requirements.

File: README.md [hunk 7]
```diff
@@ -12,32 +15,59 @@ Thoth is deployed to the following [environments](docs/environments.md).
...
+### Test a pull request
 
-### pushing a release to production
+### Push Release to Production
 
 ### Developing a new component
 
+Developing a new component under thoth-station:
+
+- Please make use of the [template-project](https://github.com/thoth-station/template-project) it has basic requirements. Github Allow users to create new repository with template projects.
```
Comment (pacospace) [inline README.md:35]: add `,`  -> `,  it has basic requirements.` 

use `allows` -> `Github allows users to create new repository with template projects.`

File: README.md [hunk 8]
```diff
@@ -12,32 +15,59 @@ Thoth is deployed to the following [environments](docs/environments.md).
...
+  If your are a new member of Team Thoth and need a repository on Thoth-station. Please create an issue in Thoth-application.
```
Comment (pacospace) [inline README.md:36]: `If you are a new member of Team Thoth and you need a repository on [thoth-station](https://github.com/thoth-station), please create an issue in [thoth-application](https://github.com/thoth-station/thoth-application).

File: README.md [hunk 9]
```diff
@@ -12,32 +15,59 @@ Thoth is deployed to the following [environments](docs/environments.md).
...
+
+- Once the Developing work is done on the repository, please make use of the Thoth-bots for smooth release. More information on updating the configuration of the Thoth-bots [click here](https://github.com/AICoE/aicoe-ci#setting-aicoe-ci-on-github-organizationrepository).
+
+- Bots also allow user to configure them to auto update the deployment of the component in thoth-application. For this firstly, we have to setup a component for the application.
+  The component must be created based on the description of the project. Following the kustomize methods the component should have two sub-directory **base** and **overlays**, which is required by argocd applications for deployment.
+  Base would contain files which would be deployed across all environments(Prod/Stage/Test), be sure to add all relevant files. (Note: If you feel a file configuration seems like a environment specific thing then it definitely doesn't belong in the base ).
```
Comment (pacospace) [inline README.md:42]: **base/**

File: README.md [hunk 10]
```diff
@@ -12,32 +15,59 @@ Thoth is deployed to the following [environments](docs/environments.md).
...
+  Overlays would contains more sub-directories based on environments (Prod/Stage/Test), please follow the pattern of other component. Add the relevant files to the overlays, you are working on.
```
Comment (pacospace) [inline README.md:43]: **overlays/**

Comment (pacospace) [inline README.md:43]: `Add the relevant files to the overlays you are working on.`

File: README.md [hunk 11]
```diff
@@ -12,32 +15,59 @@ Thoth is deployed to the following [environments](docs/environments.md).
...
+
+- Once the component contains all information requirement, we have to add an argo-cd application configuration for the component to be supported by argocd.
```
Comment (pacospace) [inline README.md:45]: `Once the component contains all information required` 

`argocd` is used in general, or should be argo-cd?

File: README.md [hunk 12]
```diff
@@ -12,32 +15,59 @@ Thoth is deployed to the following [environments](docs/environments.md).
...
+  For Stage and Test: Please make an Pull request to [aicoe-cd](https://github.com/AICoE/aicoe-cd) , Similar to this [PR](https://github.com/AICoE/aicoe-cd/pull/136/files)
```
Comment (pacospace) [inline README.md:46]: For Stage and Test: Please make a Pull Request to [aicoe-cd](https://github.com/AICoE/aicoe-cd), for one example you can refer to this [PR](https://github.com/AICoE/aicoe-cd/pull/136/files).

File: README.md [hunk 13]
```diff
@@ -12,32 +15,59 @@ Thoth is deployed to the following [environments](docs/environments.md).
...
+  For Production: Please make a Pull Request to [argo-apps](https://github.com/operate-first/argocd-apps), Similar to this [PR](https://github.com/operate-first/argocd-apps/pull/20)
```
Comment (pacospace) [inline README.md:47]: `, similar to`

File: README.md [hunk 14]
```diff
@@ -12,32 +15,59 @@ Thoth is deployed to the following [environments](docs/environments.md).
...
- - Make sure to contribute the encrypted **secrets.enc.yaml** .
+
+- First, create a `secrets.yaml` at the root level. (git would ignore the file)
+- Then run sops command to encrypt them to the desired application subdirectory. Example: `sops -e secrets.yaml > {component}/overlays/test/secrets.enc.yaml`
+- Make sure to contribute the encrypted **secrets.enc.yaml** .
 
 ## Open Questions
 
-### how do we tag release
+- What should i do if i don't know how to deploy my application?
```
Comment (pacospace) [inline README.md:62]: `What should I do if I don't know how to deploy my application?`

File: README.md [hunk 15]
```diff
@@ -12,32 +15,59 @@ Thoth is deployed to the following [environments](docs/environments.md).
...
+
+  we all are happy to help, open an issue with details of the application.
```
Comment (pacospace) [inline README.md:64]: `We all are happy to help, open an issue with details of the application.`

File: README.md [hunk 16]
```diff
@@ -12,32 +15,59 @@ Thoth is deployed to the following [environments](docs/environments.md).
...
+  Team would guide and help you with all the rest
```
Comment (pacospace) [inline README.md:65]: missing `.` at the end

File: README.md [hunk 17]
```diff
@@ -12,32 +15,59 @@ Thoth is deployed to the following [environments](docs/environments.md).
...
+
+- How do we tag release ?
+
+  Make use of the Thoth-Bots Kebechet and AICoE-CI , which helps application to release as container image or to pypi based on the nature of the application.
+
+- How do we know application is deployed in Thoth Component and running ?
```
Comment (pacospace) [inline README.md:71]: `How do we know application is deployed in Thoth Component and running ? not sure what is Thoth Component in the question, should be:

`How do we know Thoth component application is deployed in cluster and running ?

**Review by pacospace — COMMENTED** [review]
few reviews, thanks for documenting this!  💯

**Comment by harshad16** [comment]
> few more reviews, thanks for documenting this!

Thank you @pacospace  so much for such a thorough review . :100:

File: README.md [hunk 18]
```diff
@@ -12,32 +15,59 @@ Thoth is deployed to the following [environments](docs/environments.md).
...
+
+- Once the Developing work is done on the repository, please make use of the Thoth-bots for a smooth release. More information on updating the configuration of the Thoth-bots [click here](https://github.com/AICoE/aicoe-ci#setting-aicoe-ci-on-github-organizationrepository).
+
+- Bots also allow a user to configure them to auto-update the deployment of the component in thoth-application. For this firstly, we have to set up a component for the application.
+  The component must be created based on the description of the project. Following the kustomize methods, the component should have two sub-directory **base/** and **overlays/**, which is required by argocd applications for deployment.
+  The **base** would contain files that would be deployed across all environments(Prod/Stage/Test), be sure to add all relevant files. (Note: If you feel a file configuration seems like an environment-specific thing then it doesn't belong in the base ).
+  Overlays would contain more sub-directories based on environments (Prod/Stage/Test), please follow the pattern of other components. Add the relevant files to the overlays, you are working on.
```
Comment (pacospace) [inline README.md:43]: **overlays/**

Comment (pacospace) [reply README.md:43]: I would also separate the description of base and overlays instead of having one huge paragraph. For readability, WDYT?

Comment (harshad16) [reply README.md]: I tried to separate them in https://github.com/thoth-station/thoth-application/pull/740/commits/d30d56832a7812bbd59f9002e6759435b9b63f69
does this look good from a readability point of view?

Comment (pacospace) [reply README.md]: very nice!! much better indeed! Let's get this amazing doc in 💯

**Review by pacospace — APPROVED** [review]
Thanks, very nice!! 🥇

**Pull request merged**